### DISCUSSION TITLE
version solving failed

### DISCUSSION THREAD

**Discussion by ltung** [TheAlphamerc/flutter_twitter_clone#178] [General]
I downloaded the source code toay (6/3/2021) and execute: 

flutter run -d emulator-5554

It gave me the error message below. I don't know how to solve it. Anybody has an idea? Thanks.

-------------------------
The current Dart SDK version is 2.13.1.

Because no versions of freezed match >0.12.7 <0.13.0 and freezed 0.12.7 depends on analyzer >=0.40.4 <1.0.0, freezed ^0.12.7 requires analyzer >=0.40.4 <1.0.0.
Because test >=1.16.6 depends on analyzer ^1.0.0 and test >=1.9.3 <1.15.4 depends on analyzer >=0.36.0 <0.40.0, test >=1.9.3 <1.15.4-∞ or >=1.16.6 requires analyzer >=0.36.0 <0.40.0 or ^1.0.0.
Thus, freezed ^0.12.7 is incompatible with test >=1.9.3 <1.15.4-∞ or >=1.16.6.
And because test >=1.3.0 <1.12.0 depends on boolean_selector ^1.0.0 and test >=1.16.0-nullsafety.8 <1.16.0-nullsafety.18 depends on stream_channel >=2.1.0-nullsafety <2.1.0, if freezed ^0.12.7 and test >=1.3.0 <1.15.4-∞ or >=1.16.0-nullsafety.8 <1.16.0-nullsafety.18 or >=1.16.6 then boolean_selector ^1.0.0 or stream_channel >=2.1.0-nullsafety <2.1.0.
And because test >=1.16.0-nullsafety <1.16.0-nullsafety.8 requires SDK version >=2.10.0-0 <2.12.0 and test <1.3.0 requires SDK version >=1.8.0 <2.0.0-∞, if freezed ^0.12.7 and test <1.15.4-∞ or >=1.16.0-nullsafety <1.16.0-nullsafety.18 or >=1.16.6 then boolean_selector ^1.0.0 or stream_channel >=2.1.0-nullsafety <2.1.0.
And because test_api >=0.2.19 depends on both boolean_selector ^2.1.0 and stream_channel ^2.1.0, one of freezed ^0.12.7 or test <1.15.4-∞ or >=1.16.0-nullsafety <1.16.0-nullsafety.18 or >=1.16.6 or test_api >=0.2.19 must be false.
And because test >=1.16.0-nullsafety.19 <1.16.6 depends on test_api 0.2.19 and test >=1.15.5 <1.16.0-nullsafety depends on test_api 0.2.18+1, one of freezed ^0.12.7 or test_api >0.2.19 or test <1.15.4-∞ or >=1.15.5 <1.16.0-nullsafety.18 or >=1.16.0-nullsafety.19 must be false.
And because test >=1.15.3 <1.15.5 depends on test_api 0.2.18 and test >=1.16.0-nullsafety.18 <1.16.0-nullsafety.19 depends on test_api 0.2.19-nullsafety.7, one of freezed ^0.12.7 or test_api >0.2.19 or test any must be false.
And because every version of flutter_test from sdk depends on test_api 0.3.0 and flutter_twitter_clone depends on freezed ^0.12.7, test is incompatible with flutter_test from sdk.
So, because flutter_twitter_clone depends on both test any and flutter_test any from sdk, version solving failed.
Running "flutter pub get" in flutter_twitter_clone...
pub get failed (1; So, because flutter_twitter_clone depends on both test any and flutter_test any from sdk, version
solving failed.)
-----------------------

**Comment by TheAlphamerc** [comment]
Hi @ltung , Please check this [thread](https://github.com/TheAlphamerc/flutter_twitter_clone/issues/172) and let me know if it helps.

**Reply by ltung** [reply]
Yes, the downgrading works. Thanks.
